### PR TITLE
Added date parsing module

### PR DESCRIPTION
**Pull request by rmdort**
Let me know if i am going in the right direction

Todo

1. Documentation...
2. List of all supported date patterns
3. Add tests

### CODE REVIEW

**Comment by gunthercox** [comment]
@rmdort, this looks fantastic so far.

File: chatterbot/utils/parsing.py [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+
+"""
+The first step is the tag the text with <TIMEX>
+1. Extract => The meeting is 25 minutes from now => The meeting is <TIMEX>25 minutes from now<TIMEX>
+2. Convert to datetime
+"""
+
+"""
+Regular expression matches
+https://github.com/nltk/nltk_contrib/blob/master/nltk_contrib/timex.py
```
Comment (gunthercox) [inline chatterbot/utils/parsing.py]: We need to be careful with licensing here. NLTK is [Apache License 2.0](http://www.apache.org/licenses/LICENSE-2.0)

**Comment by rmdort** [comment]
Actually I don't use nltk or MX date in the parser, I will remove the references.

By the way, aren't u using nltk internally to remove stopwords ?

**Comment by gunthercox** [comment]
@rmdort NLTK is used internally in this project, but as a dependency this is installed separately though the python package manager. This is different than redistributing code from anther project.

**Comment by rmdort** [comment]
Ah ok, I see. 

What do you think i should do? I did borrow most of the code from nltk_contrib, but made a lot of modifications to match patterns.  Will a rewrite work, maybe a restructure?

File: chatterbot/utils/parsing.py (hunk 1 above)
Comment (kevin-brown) [inline chatterbot/utils/parsing.py]: Apache 2.0 requires you to, at the very minimum (probably more to it), include the license text in the files which are licensed under Apache 2.0.

You would be better off optionally depending on NLTK for any adapters which need date parsing support.

**Comment by rmdort** [comment]
Just did a total overhaul of the date parser from scratch. Its much easier to add new pattern now. No external dependencies nor any nltk code being used.

I will add more regex patterns asap.

**Comment by gunthercox** [comment]
Looks good :+1: Thank you!

**Comment by rmdort** [comment]
Done. If i missed out any date patterns let me know. Otherwise its ready

File: chatterbot/utils/parsing.py [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+            [-\s] # One or more space
+            (?P<month>%s) # Matches any month name
+            [-\s] # Space
+            (?P<year>\d{4}) # Year
+            ((\s|,\s|\s(%s))?\s*(%s))?
+        )
+        '''% (day_names, re_ordinal, month_names, re_separator, re_time),
+        (re.VERBOSE | re.IGNORECASE)
+        ),
+        lambda (m, base_date): datetime(
```
Comment (rmdort) [inline chatterbot/utils/parsing.py]: Sure. I am going to add support for other languages and will push the changes soon.

**Pull request merged**